File: core/src/main/java/com/aaronicsubstances/code/augmentor/core/tasks/ProcessCodeContext.java
package com.aaronicsubstances.code.augmentor.core.tasks;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

import com.aaronicsubstances.code.augmentor.core.models.GeneratedCode;
import com.aaronicsubstances.code.augmentor.core.models.SourceFileAugmentingCode;

public class ProcessCodeContext {
    private final Map<String, Object> globalScope = new HashMap<>();
    private final Map<String, Object> fileScope = new HashMap<>();
    private SourceFileAugmentingCode fileAugCodes;
    private int augCodeIndex;

    // used by Groovy script.
    public GeneratedCode newGenCode() {
        return new GeneratedCode(new ArrayList<>());
    }

    public Map<String, Object> getGlobalScope() {
        return globalScope;
    }

    public Map<String, Object> getFileScope() {
        return fileScope;
    }

    public SourceFileAugmentingCode getFileAugCodes() {
        return fileAugCodes;
    }

    void setFileAugCodes(SourceFileAugmentingCode fileAugCodes) {
        this.fileAugCodes = fileAugCodes;
    }

    public int getAugCodeIndex() {
        return augCodeIndex;
    }

    void setAugCodeIndex(int augCodeIndex) {
        this.augCodeIndex = augCodeIndex;
    }
}

File: supporting-packages/php7/codeaugmentor_support/src/models/ProcessCodeContext.php
<?php declare(strict_types=1);
namespace aaronicsubstances\codeaugmentor_support\models;

class ProcessCodeContext {

   public $globalScope = [];
   public $fileScope = [];
   public $fileAugCodes;
   public $augCodeIndex = 0;
   public $srcFile;
   
    public function newGenCode() {
        return new GeneratedCode;
    }

    public function newContent(string $content, bool $exactMatch = FALSE) : ContentPart    {
        return new ContentPart($content, $exactMatch);
    }
    
    public function __toString() {
        return get_class($this) . ": " . $this->srcFile;
    }
}